### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by phucynwa**

### CODE REVIEW

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -2,3 +2,66 @@
...
+      display: block;
+      margin-bottom: 3px;
+      line-height: 1;
+    }
+    h1 {
+      font-size: 1.4em;
+      text-align: left;
+      letter-spacing: -1px;
+      margin-bottom: 3px;
+      margin-top: 0px;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: You can combine line 28 and 27 into one

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
-  get "static_pages/home"
   root "static_pages#home"
+  get "static_pages/home"
+  get "/signup", to: "users#new"
+  post "/signup", to: "users#create"
+  resources :users
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Specify the action here.

File: config/locales/en.yml [hunk 3]
```diff
@@ -34,3 +34,28 @@ en:
...
+      confirm_password: "Confirm Your Password"
+    create:
+      sign_up_successful: "Sign up successful !"
+  shared:
+    error_messages:
+      form_error_warning:
+        zero: "none"
+        one: "The form contains 1 error"
+        other: "The form contains %{count} errors"
+
```
Comment (mdmahmudurrahman) [inline config/locales/en.yml]: Please remove this empty line

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
-  get "static_pages/home"
   root "static_pages#home"
+  get "static_pages/home"
+  get "/signup", to: "users#new"
+  post "/signup", to: "users#create"
+  resources :users do
+    get "/:id", action: :show
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Why you added id here?

Comment (phucynwa) [reply config/routes.rb]: @mdmahmudurrahman id of user, i think that is what you want me to edit.

Comment (mdmahmudurrahman) [reply config/routes.rb]: `resources :users, only: %i(:new, :create :show)`

Comment (phucynwa) [reply config/routes.rb]: @mdmahmudurrahman yeah, i fixed.

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -2,3 +2,65 @@
...
+  }
+}
+
+.field_with_errors {
+  @extend .has-error;
+  .form-control {
+    color: $flag-red;
+  }
+}
+.alert {
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Please add a line break

File: app/views/users/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<%= form_for @user do |f| %>
+  <%= render "shared/error_messages", object: f.object %>
+  <%= f.text_field :name, class: "form-control", placeholder: t(".name") %>
+
+  <%= f.email_field :email, class: "form-control", placeholder: t(".email")  %>
```
Comment (mdmahmudurrahman) [inline app/views/users/_form.html.erb]: Please remove redundant white space

File: app/views/users/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+  <%= f.phone_field :phone, class: "form-control", placeholder: t(".phone")  %>
```
Comment (mdmahmudurrahman) [inline app/views/users/_form.html.erb]: Same here before closing the erb tag

File: app/views/users/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+  <%= f.password_field :password, class: "form-control",
+    placeholder: t(".password") %>
+
+  <%= f.password_field :password_confirmation, class: "form-control",
+    placeholder: t(".confirm_password")  %>
```
Comment (mdmahmudurrahman) [inline app/views/users/_form.html.erb]: Same here

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -24,7 +24,7 @@
       <div class="row">
         <div class="col-md-2 col-sm-6 col-xs-6">
           <div id="logo">
-            <a href="index.html"><%= raw t(".web_name") %></a>
+            <a href="/"><%= raw t(".web_name") %></a>
```
Comment (hungnh103) [inline app/views/layouts/_header.html.erb]: use Rails helper `link_to` instead of raw `HTML`

File: app/assets/stylesheets/custom.scss [hunk 10]
```diff
@@ -2,3 +2,66 @@
...
+    span {
+      display: block;
+      margin-bottom: 3px;
+      line-height: 1;
+    }
+    h1 {
+      font-size: 1.4em;
+      text-align: left;
+      letter-spacing: -1px;
+      margin-bottom: 0 0 3px 0;
```
Comment (hungnh103) [inline app/assets/stylesheets/custom.scss]: extract this line
it can be written shorter

File: app/assets/stylesheets/custom.scss [hunk 11]
```diff
@@ -2,3 +2,66 @@
...
+
+.link {
+  color: $flag-red;
+}
+
+#error_explanation {
+  color: red;
+  ul {
+    color: red;
+    margin: 0 0 30px 0;
```
Comment (hungnh103) [inline app/assets/stylesheets/custom.scss]: as above comment

**Pull request merged**